### ISSUE TITLE
Older client's dot release image is tagged as latest

### ISSUE THREAD

**Issue by dsimansk** [knative/test-infra#2593]
I've noticed that knative/client dot release `v0.18.4` that was produced a couple of hours after `v0.19.1` is tagged with `latest`.

I'd expect that `latest` tag is moving only on current dot releases, e.g `v0.19.1,v0.19.2,...`, excluding the previous versions.

Is it something we can fix in client's release script itself? Or a broader issue?

In addition created timestamp on image `v0.19.1` is a bit off `Jan 1, 1970`. Might it be the cause?

https://console.cloud.google.com/gcr/images/knative-releases/GLOBAL/knative.dev/client/cmd/kn?gcrImageListsize=30&pli=1

![Screenshot from 2020-11-26 15-55-12](https://user-images.githubusercontent.com/5637792/100365639-0eb7ec80-3000-11eb-861d-8168041c13d9.png)

/cc @navidshaikh @rhuss @maximilien

**Comment by rhuss** [comment]
Good point, but I have no idea how the tagging works during an auto-release. @chizhg , any ideas ?

**Comment by rhuss** [comment]
/remove-lifecycle stale

@chizhg is there a way to influence to which image version the `latest` tag should go ?

**Comment by rhuss** [comment]
/remove-lifecycle stale

ping @chizhg

**Comment by chizhg** [comment]
The `latest` tag is added by the `ko resolve` command by default - https://github.com/google/ko/blob/23ecf4753b4f59fba8d370ed554da7580bfb4082/pkg/commands/options/publish.go#L71. You can add some logics around https://github.com/knative/client/blob/main/hack/release.sh#L45 if you want to change this behavior.

The reason that the images are all created in 1970s is explained in https://github.com/google/ko#why-are-my-images-all-created-in-1970

**Comment by dsimansk** [comment]
@chizhg thanks for the thorough explanation. I'll take a look at patching the release script. IMO this issue can be closed as resolved/explained.

/close

**Comment by dprotaso** [comment]
I notice latest kn version aren't being tagged

https://cloud-native.slack.com/archives/C04LY4SKBQR/p1679071860118629
